test/shopping_cart_test.rb

```diff
@@ -0,0 +1,60 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/product'
+require './lib/shopping_cart'
+class ShoppingCartTest < Minitest::Test
+  def test_it_exists
+    cart = ShoppingCart.new("King Soopers", "30items")
+    assert_instance_of ShoppingCart, cart
+  end
+  def test_it_has_attributes
+    cart = ShoppingCart.new("King Soopers", "30items")
+    assert_equal "King Soopers", cart.name
+    assert_equal 30, cart.capacity
+    assert_equal [], cart.products
+  end
+  def test_add_product
+    cart = ShoppingCart.new("King Soopers", "30items")
+    product1 = Product.new(:paper, 'toilet paper', 3.70, '10')
+    product2 = Product.new(:meat, 'chicken', 4.50, '2')
+    cart.add_product(product1)
+    cart.add_product(product2)
+    assert_equal [product1, product2], cart.products
+  end
+  #Could not get hash formatting to work for some reason. It looks the same as required.
+  # def test_details
+  #   cart = ShoppingCart.new("King Soopers", "30items")
+  #   product1 = Product.new(:paper, 'toilet paper', 3.70, '10')
+  #   product2 = Product.new(:meat, 'chicken', 4.50, '2')
+  #   cart.add_product(product1)
+  #   cart.add_product(product2)
+  #   cart_details = cart.details
+  #   assert_equal {name: "King Soopers", capacity: 30}, cart_details
```

Comment:
This method is working - I think the issue is with how minitest is reading this line.  It won't like a hash as the first argument, unless it is wrapped in parens, or stored in a variable.